modules/post/multi/gather/enum_software_versions.rb

```diff
@@ -0,0 +1,139 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Multiplatform Installed Software Version',
+        'Description' => %q{
+          This module, when run against a compromised machine, will gather details on all installed software,
+          including their versions and if available, when they were installed, and will save it into a loot file for later use.
+          Users can then use this loot file to determine what additional vulnerabilites may affect the target machine.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'gwillcox-r7' ],
+        'Platform' => %w[win linux osx bsd solaris android],
+        # 'Platform'      => %w{ android osx win linux bsd solaris },
+        'SessionTypes' => [ 'meterpreter', 'shell' ]
+      )
+    )
+  end
+
+  def store_linux_loot(listing)
+    file = store_loot('host.linux.software.versions', 'text/plain', session, listing, 'installed_software.txt', 'Installed Software and Versions')
+    print_good("Stored information about the installed products to the loot file at #{file}")
+  end
+
+  # Run Method for when run command is issued
+  def run
+    case session.platform
+    when 'windows'
+      listing = cmd_exec('wmic product get Name, Description, Version, InstallDate', nil, 6000)
+      unless listing =~ /Description/
```

Comment:
Using `include?` is usually preferred for a simple string match. It is implemented as a direct string lookup, which is faster than regex.
```suggestion
      unless listing.include?('Description')
```

Reply:
Fixed in next commit and good point on the speed @cdelafuente-r7, didn't think of that but very true 👍

Reply:
`include?` is preferred, but there's a gotcha in that it is accessible only on string objects. `cmd_exec` can return `nil` or `raise`.

It's been a while since I've looked, but I'm pretty sure this was never "fixed".

As such, it is generally best to call `to_s` on the results of the `cmd_exec`.

```ruby
listing = cmd_exec('...').to_s
```

Reply:
Good point @bcoles, going to cast the results of the cmd_exec calls to a string using `.to_s` as you recommended since this should prevent us crashing if it returns `nil`.

Reply:
Should be resolved with the next commit.